src/myrtlespeech/model/rnn.py

```diff
@@ -11,6 +13,13 @@ class RNNType(IntEnum):
     BASIC_RNN = 2
 
 
+RNNState = TypeVar("RNNState", torch.Tensor, Tuple[torch.Tensor, torch.Tensor])
```

Comment:
`TypeVar` is used to define possibly constrained generic types and function definitions. `NewType` is used to define essentially a type alias. 

`RNNState` and `RNNInput` are constrained generic types as the type passed in to `forward` is the type returned. RNNLengths is not generic and is instead a type alias.

I think these definitions should therefore be something similar to that below. Also note that they appear to be rendered by Sphinx without any of the right hand side. Comments can be added to them by including a docstring *below* the definition.

```suggestion
RNNLengths = NewType("RNNLengths", torch.Tensor)
"""RNNLengths comment TODO"""

RNNState = TypeVar("RNNState", torch.Tensor, Tuple[torch.Tensor, torch.Tensor])
"""RNNState comment TODO"""

RNNInput = TypeVar(
    "RNNInput",
    torch.Tensor,
    Tuple[torch.Tensor, Optional[RNNState]]
)
"""RNNInput comment TODO"""
```

More generally, moving the `RNNLengths` `NewType` to just `Lengths` or `Steps` or similar and using that as the type in every module that accepts lengths as input would be a good idea! Although perhaps create this as an issue/on the backlog and we'll do it in a future PR?

Reply:
If comments are added to `RNNLengths` etc then the docstrings for `RNN` can be shortened a lot! e.g. no need to repeat what each of them are!

Reply:
I've added your lengths suggestion to the backlog and moved `RNNLengths` -> `Lengths` in prep for this.

Reply:
Ok thanks for the comparison of `TypeVar` vs `NewVar`. I updated `RNNInput -> RNNData` to and made it a TypeVar to reflect this (some mypy ignoring on the definition was required). 

I've tried to add the line:
```python
Lengths = NewType("Lengths", torch.Tensor)
```
and while I take the point that this is correct usage, unless I've made a mistake, I think this has two disadvantages vs:
```python
Lengths = TypeVar("Lengths", bound=torch.Tensor)
```

1) The `NewType` version runs but mypy throws:
```shell
src/myrtlespeech/model/rnn.py:38: error: Argument 2 to NewType(...) must be subclassable (got "Any")
src/myrtlespeech/model/rnn.py:139: error: Invalid type "myrtlespeech.model.rnn.Lengths"
```
i.e. I think it is struggling to subclass `torch.Tensor` for some reason. This is ok, we can hack around this by adding:
```python
class _TorchTensor(torch.Tensor):
    """Subclass of :py:class:`torch.Tensor` that is itself subclassable."""
Lengths = NewType("Lengths", _TorchTensor)
```
and mypy is happy. 

2) The `NewType` version displays the forward signature weirdly in the docs as:
```python
forward(x: Tuple[RNNData, NewType.<locals>.new_type]) → Tuple[RNNData, NewType.<locals>.new_type]
```
instead of for the `TypeVar` version:
```python
forward(x: Tuple[RNNData, Lengths]) → Tuple[RNNData, Lengths]
```

FYI - I also tried Aliasing directly with:
```python
Lengths=torch.Tensor
```
...but this is the worst of all three options as Sphinx then completely ignores the attribute string and just gives `Lengths is an alias of torch.Tensor`.

Reply:
^ As such I've left as `TypeVar("Lengths", bound=torch.Tensor)` but v happy to change if you prefer?

Reply:
Update: to fix the `check-docstring-first` ci errors, all attribute strings are now in `#: ...` form.

Reply:
Okay, thanks for the investigation :-) Let's go with what works